src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -440,3 +473,144 @@ def _get_install_info(cmd,
+                             connected_registry,
+                             add_actions_set,
+                             remove_actions_set,
+                             msg=None,
+                             description=None):
+    scope_map_client = cf_acr_scope_maps(cmd.cli_ctx)
+    sync_token = get_token_from_id(cmd, connected_registry.parent.sync_properties.token_id)
+    sync_scope_map = get_scope_map_from_id(cmd, sync_token.scope_map_id)
+    sync_scope_map_name = sync_scope_map.name
+    current_actions_set = set(sync_scope_map.actions)
+    final_actions_set = current_actions_set.union(add_actions_set).difference(remove_actions_set)
+    if final_actions_set == current_actions_set:
+        return None
+    current_actions = list(final_actions_set)
+    logger.warning(msg)
+    return scope_map_client.update(
+        resource_group_name,
+        registry_name,
+        sync_scope_map_name,
+        description,
+        current_actions
+    )
+
+
+def _get_scope_map_actions_set(repos, actions):
+    for i, repo_name in enumerate(repos):
+        repos[i] = [repo_name] + actions
+    return set(parse_scope_map_actions(repos))
+
+
+def acr_connected_registry_repo(cmd,
+                                client,
+                                connected_registry_name,
+                                registry_name,
+                                add_repos=None,
+                                remove_repos=None,
+                                resource_group_name=None):
+    if not (add_repos or remove_repos):
+        raise CLIError('No repository permissions to update.')
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    add_repos_set = set(add_repos) if add_repos is not None else set()
+    remove_repos_set = set(remove_repos) if remove_repos is not None else set()
+    duplicate_repos = set.intersection(add_repos_set, remove_repos_set)
+    if duplicate_repos:
+        errors = sorted(map(lambda action: action[action.rfind('/') + 1:], duplicate_repos))
+        raise CLIError(
+            'Update ambiguity. Duplicate repository names were provided with ' +
+            '--add and --remove arguments.\n{}'.format(errors))
+
+    connected_registry_list = list(client.list(resource_group_name, registry_name))
+    family_tree, target_connected_registry = _get_family_tree(connected_registry_list, connected_registry_name)
+    if target_connected_registry is None:
+        raise CLIError("Connected registry '{}' doesn't exist.".format(connected_registry_name))
+
+    # remove repo permissions from connected registry lineage.
+    remove_actions = REPO_SCOPES_BY_MODE[ConnectedRegistryModes.REGISTRY.value]
+    if remove_repos is not None:
+        remove_repos_txt = ", ".join(remove_repos)
+        remove_repos_set = _get_scope_map_actions_set(remove_repos, remove_actions)
+        lineage = _get_lineage(family_tree, target_connected_registry.id)
```

Comment:
is it to update all children?

Reply:
I'm asking as I usually use `descendants` vs `ancestors` in a tree structure?

Reply:
yes, lineage is for all the way to the leaves. I thought lineage was more descriptive. So I changed the naming everywhere to lineage. But I can roll back if you think descendants is better.